dll/win32/shell32/CSendToMenu.cpp

```diff
@@ -0,0 +1,518 @@
+        return E_FAIL;
+
+    return m_pSite->QueryInterface(riid, ppvSite);
+}
+
+STDMETHODIMP
+CSendToMenu::QueryContextMenu(HMENU hMenu,
+                              UINT indexMenu,
+                              UINT idCmdFirst,
+                              UINT idCmdLast,
+                              UINT uFlags)
+{
+    TRACE("%p %p %u %u %u %u\n", this,
+          hMenu, indexMenu, idCmdFirst, idCmdLast, uFlags);
+
+    WCHAR wszSendTo[64];
+    if (!LoadStringW(shell32_hInstance, IDS_SENDTO,
+                     wszSendTo, _countof(wszSendTo)))
+    {
+        ERR("IDS_SENDTO\n");
+        return E_FAIL;
+    }
+
+    HMENU hSubMenu = CreateMenu();
+    if (!hSubMenu)
+    {
+        ERR("CreateMenu\n");
+        return E_FAIL;
+    }
+
+    UINT cItems = InsertSendToItems(hSubMenu, idCmdFirst, 0);
+
+    MENUITEMINFOW mii;
+    ZeroMemory(&mii, sizeof(mii));
+    mii.cbSize = sizeof(mii);
+    mii.fMask = MIIM_TYPE | MIIM_ID | MIIM_STATE | MIIM_SUBMENU;
+    mii.fType = MFT_STRING;
+    mii.wID = -1;
+    mii.dwTypeData = wszSendTo;
+    mii.cch = wcslen(mii.dwTypeData);
+    mii.fState = MFS_ENABLED;
+    mii.hSubMenu = hSubMenu;
+    if (!InsertMenuItemW(hMenu, indexMenu, TRUE, &mii))
+    {
+        ERR("InsertMenuItemW\n");
+        return E_FAIL;
+    }
+
+    HMENU hOldSubMenu = m_hSubMenu;
+    m_hSubMenu = hSubMenu;
+    DestroyMenu(hOldSubMenu);
+
+    return MAKE_HRESULT(SEVERITY_SUCCESS, 0, cItems);
+}
+
+STDMETHODIMP
+CSendToMenu::InvokeCommand(LPCMINVOKECOMMANDINFO lpici)
+{
+    HRESULT hr = E_FAIL;
+
+    WORD idCmd = LOWORD(lpici->lpVerb);
+    TRACE("idCmd: %d\n", idCmd);
+
+    SENDTO_ITEM *pItem = FindItemFromIdOffset(idCmd);
+    if (pItem)
+    {
+        hr = DoSendToItem(pItem, lpici);
+    }
+    else
+    {
+        ERR("FindItemFromIdOffset\n");
+    }
+
+    TRACE("CSendToMenu::InvokeCommand %x\n", hr);
+    return hr;
+}
+
+STDMETHODIMP
+CSendToMenu::GetCommandString(UINT_PTR idCmd,
+                              UINT uType,
+                              UINT *pwReserved,
+                              LPSTR pszName,
+                              UINT cchMax)
+{
+    FIXME("%p %lu %u %p %p %u\n", this,
+          idCmd, uType, pwReserved, pszName, cchMax);
+
+    return E_NOTIMPL;
+}
+
+STDMETHODIMP
+CSendToMenu::HandleMenuMsg(UINT uMsg, WPARAM wParam, LPARAM lParam)
+{
+    return S_OK;
+}
+
+STDMETHODIMP
+CSendToMenu::HandleMenuMsg2(UINT uMsg, WPARAM wParam, LPARAM lParam,
+                            LRESULT *plResult)
+{
+    UINT cxSmall = GetSystemMetrics(SM_CXSMICON);
+    UINT cySmall = GetSystemMetrics(SM_CYSMICON);
+
+    switch (uMsg)
+    {
+    case WM_MEASUREITEM:
+        {
+            MEASUREITEMSTRUCT* lpmis = reinterpret_cast<MEASUREITEMSTRUCT*>(lParam);
+            if (!lpmis || lpmis->CtlType != ODT_MENU)
+                break;
+
+            if (lpmis->itemWidth < (UINT)GetSystemMetrics(SM_CXMENUCHECK))
+                lpmis->itemWidth = GetSystemMetrics(SM_CXMENUCHECK);
```

Comment:
Two calls of the same function with same parameters?

Reply:
Fixed.